### PR TITLE
use CustomMuiDialog and get initial name from directory-server

### PR DESCRIPTION
**Pull request by achour94**
use CustomMuiDialog and UniqueNameInput instead of old implementation.
generate intial name from the directory server

### CODE REVIEW

File: src/components/dialogs/copy-to-script-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * @param open Is the dialog open ?
+ * @param onClose Event to close the dialog
+ * @param onClick Function to call to perform copy
+ * @param currentName Name before renaming
+ * @param title Title of the dialog
+ * @param directoryUuid Directory uuid of the list or filter to copy
+ * @param elementType Type of the element to copy
+ * @param handleError Function to call to handle error
+ */
+const CopyToScriptDialog: React.FC<CopyToScriptDialogProps> = ({
```
Comment (flomillot) [inline src/components/dialogs/copy-to-script-dialog.tsx]: I think we decided to use `FunctionalComponent` everywhere and not `FC`.

File: src/components/dialogs/copy-to-script-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    const nameError = errors[NAME];
+    const isValidating = errors.root?.isValidating;
+
+    const onSubmit = (data: { [NAME]: string; [key: string]: any }) => {
+        onClick(id, data[NAME]);
+    };
+
+    const handleClose = () => {
+        reset(emptyFormData);
```
Comment (flomillot) [inline src/components/dialogs/copy-to-script-dialog.tsx]: I don't think it's necessary

File: src/components/dialogs/copy-to-script-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+ * @param title Title of the dialog
+ * @param directoryUuid Directory uuid of the list or filter to copy
+ * @param elementType Type of the element to copy
+ * @param handleError Function to call to handle error
+ */
+const CopyToScriptDialog: React.FC<CopyToScriptDialogProps> = ({
+    id,
+    open,
+    onClose,
+    onClick,
```
Comment (flomillot) [inline src/components/dialogs/copy-to-script-dialog.tsx]:
```suggestion
    onValidate,
```

File: src/components/dialogs/copy-to-script-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    const {
+        reset,
+        formState: { errors },
+        setValue,
+    } = methods;
+
+    const nameError = errors[NAME];
+    const isValidating = errors.root?.isValidating;
+
+    const onSubmit = (data: { [NAME]: string; [key: string]: any }) => {
```
Comment (flomillot) [inline src/components/dialogs/copy-to-script-dialog.tsx]: Why `[key: string]: any` ?
If you want you can do like it's done for CreateCaseDialog with `IFormData` (you can remove the `I` before)

File: src/translations/en.json [hunk 5]
```diff
@@ -27,6 +27,7 @@
   "deleteElementsFailure": "Problem encountered while deleting {stn, plural, one {an element} other {elements}}. Not deleted : {problematic}",
   "moveElementsFailure": "Problem encountered while moving {stn, plural, one {an element} other {elements}}. Not moved : {problematic}",
   "duplicateElementFailure": "An error occured while duplicating the element {itemName} : {errorMessage}",
+  "generateCopyScriptNameError" : "An error occured while generating the copy script name of the element {itemName} : {errorMessage}",
```
Comment (flomillot) [inline src/translations/en.json]: I would not set the server error message into the snack bar displayed to the user. I don't know why it's done elsewhere but for me it's not a good practice. I asked on slack if we should do this or not.

File: src/components/dialogs/copy-to-script-dialog.tsx [hunk 6]
```diff
@@ -28,31 +28,35 @@ interface CopyToScriptDialogProps {
...
-    onClick: (...args: any[]) => void;
+    onValidate: (...args: any[]) => void;
     currentName: string;
     title: string;
     directoryUuid: string;
     elementType: ElementType;
     handleError: (...args: any[]) => void;
 }
 
+interface IFormData {
```
Comment (flomillot) [inline src/components/dialogs/copy-to-script-dialog.tsx]: Please remove the `I`

**Review by flomillot — CHANGES_REQUESTED** [review]

**Review by flomillot — APPROVED** [review]

**Review by ghazwarhili — APPROVED** [review]

**Pull request merged**